dspace-api/src/main/java/org/dspace/discovery/indexobject/ItemIndexFactoryImpl.java

```diff
@@ -0,0 +1,684 @@
+    @Autowired
+    protected MetadataAuthorityService metadataAuthorityService;
+
+
+    @Override
+    public Iterator<IndexableItem> findAll(Context context) throws SQLException {
+        Iterator<Item> items = itemService.findAllUnfiltered(context);
+        return new Iterator<IndexableItem>() {
+            @Override
+            public boolean hasNext() {
+                return items.hasNext();
+            }
+
+            @Override
+            public IndexableItem next() {
+                return new IndexableItem(items.next());
+            }
+        };
+    }
+
+    @Override
+    public String getType() {
+        return IndexableItem.TYPE;
+    }
+
+    /**
+     * Build a Solr document for a DSpace Item and write the index
+     *
+     * @param context       Users Context
+     * @param indexableItem The IndexableItem Item to be indexed
+     * @throws SQLException if database error
+     * @throws IOException  if IO error
+     */
+    @Override
+    public SolrInputDocument buildDocument(Context context, IndexableItem indexableItem)
+            throws SQLException, IOException {
+        SolrInputDocument doc = super.buildDocument(context, indexableItem);
+
+        final Item item = indexableItem.getIndexedObject();
+
+        doc.addField("archived", item.isArchived());
+        doc.addField("withdrawn", item.isWithdrawn());
+        doc.addField("discoverable", item.isDiscoverable());
+        doc.addField("lastModified", SolrUtils.getDateFormatter().format(item.getLastModified()));
+
+        EPerson submitter = item.getSubmitter();
+        if (submitter != null) {
+            addFacetIndex(doc, "submitter", submitter.getID().toString(),
+                    submitter.getFullName());
+        }
+
+        List<DiscoveryConfiguration> discoveryConfigurations = SearchUtils.getAllDiscoveryConfigurations(item);
+        addDiscoveryFields(doc, context, indexableItem.getIndexedObject(), discoveryConfigurations);
+
+        //mandatory facet to show status on mydspace
+        final String typeText = StringUtils.deleteWhitespace(indexableItem.getTypeText().toLowerCase());
+        String acvalue = DSpaceServicesFactory.getInstance().getConfigurationService().getProperty(
+                "discovery.facet.namedtype." + typeText,
+                typeText + SearchUtils.AUTHORITY_SEPARATOR + typeText);
+        if (StringUtils.isNotBlank(acvalue)) {
+            addNamedResourceTypeIndex(doc, acvalue);
+        }
+
+        // write the index and close the inputstreamreaders
+        try {
+            log.info("Wrote Item: " + item.getID() + " to Index");
+        } catch (RuntimeException e) {
+            log.error("Error while writing item to discovery index: " + item.getID() + " message:"
+                    + e.getMessage(), e);
+        }
+        return doc;
+    }
+
+    @Override
+    public void addDiscoveryFields(SolrInputDocument doc, Context context, Item item,
```

Comment:
This code was just copy pasted from the following method: https://github.com/DSpace/DSpace/blob/dspace-6_x/dspace-api/src/main/java/org/dspace/discovery/SolrServiceImpl.java#L949

Documenting this would be quite a task, as we would have to look into what everything does, I believe this is better suited for follow up code refactors where we break up giant methods like this, as a method this big is a code smell.

Reply:
Ah, I didn't realize this was copied directly from old code. In that case, I agree this might be done as a separate task.